sdk/enoki/src/react.tsx

```diff
@@ -67,3 +74,45 @@ export function useAuthCallback() {
 
 	return { handled, state };
 }
+
+export const EnokiWalletContext = createContext<{
+	wallets: ReturnType<typeof registerEnokiWallets>['wallets'];
+	client: SuiClient;
+} | null>(null);
+
+export function EnokiWalletProvider({
+	children,
+	config,
+	useSuiClientContext,
+}: {
+	config: Omit<Parameters<typeof registerEnokiWallets>[0], 'client'>;
+	useSuiClientContext: () => { client: SuiClient; network: string };
```

Comment:
passing the hook through as a prop feels slightly unidiomatic even though it's valid, eeeh I could go either way on passing the hook vs `client` and `network` separately - I'm not sure what's easier/better for developers or if this doesn't make a difference at all hahaha

cc @Jordan-Mysten if you have any preference here 🫡

Reply:
Forgot about this... I would love a better way to do this.

What I want is for the provider to be able to get the network/client without introducing a direct dependency between dapp-kit and the enoki SDK.

Basically  I want this to work without having to manually manage the network/client via some external state.  Passing in the client/network would require having access to them here
```tsx
<SuiClientProvider ...>
    <EnokiWalletProvider ...>
       <App />
    </EnokiWalletProvider>
 </SuiClientProvider>
 ```

Reply:
oh sorry I totally skimmed over the actual usage haha, mmm you could make `SuiClientProvider` take a `function` children prop so you could write something the following:

```
<SuiClientProvider propblahblah>
  {({ client, network }) =>  (
    <EnokiWalletProvider client={client} network={network}>
       ....
    </EnokiWalletProvider>
  )}
</SuiClientProvider>
```

I don't know if that's any better than what you originally wrote though 🤷🏼‍♂️

Reply:
So I think passing a hook is kind of a non-starter IMO because it's just begging to break hook call order in some weird way. The dapp-kit dependency does suck too though.

A few random:
- Taking a step back, can we work the dependency of needing the client out? We should _really_ only need the network, outside of constructing transaction bytes (which we could use default providers for maybe?). Eventually Enoki will have RPC services that also solve this....
- Can we actually remove the React dependency entirely, and just make this a function you call in your application as a side-effect?
- We could bake this directly into dapp-kit?

Reply:
> Taking a step back, can we work the dependency of needing the client out? We should really only need the network, outside of constructing transaction bytes (which we could use default providers for maybe?).  
Eventually Enoki will have RPC services that also solve this....

This is an interesting question, I think we could potentially get away with just reading the network, but I think letting apps define an RPC provider once and have it work everywhere is pretty nice
> 
Can we actually remove the React dependency entirely, and just make this a function you call in your application as a side-effect?

Technically we could, but I think having this automatically work with the network switching logic in dapp-kit is something we should support

> We could bake this directly into dapp-kit?

If we are okay adding the enoki dependency to dapp-kit we can move the registration to the WalletProvider in dapp-kit